uaclient/cli.py

```diff
@@ -346,28 +346,38 @@ def _attach_with_token(
     return 0
 
 
-@assert_not_attached
-@assert_root
-def action_attach_premium(args, cfg):
+def _get_contract_token_from_cloud_identity(cfg):
+    """Detect cloud_type and request a contract token from identity info."""
     cloud_type = identity.get_cloud_type()
     if cloud_type not in ("aws",):  # TODO(avoid hard-coding supported types)
         print(
             ua_status.MESSAGE_UNSUPPORTED_PREMIUM_CLOUD_TYPE.format(
                 cloud_type=cloud_type
             )
         )
-        return 0
-    # TODO(Clean up the attach_premium flow a bit and add error handling)
+        return None
```

Comment:
Would this and the call site be simpler if we raised a `UserFacingError` with exit_code set to 0?

Reply:
(FAOD, I mean a new `UserFacingError` sub-class.)